warehouse/router/upload.go

```diff
@@ -427,6 +431,42 @@ func (job *UploadJob) run() (err error) {
 	return nil
 }
 
+func (job *UploadJob) cleanupObjectStorageFiles() error {
+	destination := job.warehouse.Destination
+	cleanupObjectStorageFiles := job.warehouse.GetBoolDestinationConfig(model.CleanupObjectStorageFilesSetting)
+	if !cleanupObjectStorageFiles {
+		return nil
+	}
+	storageProvider := whutils.ObjectStorageType(destination.DestinationDefinition.Name, destination.Config, job.upload.UseRudderStorage)
```

Comment:
We can create destination later?
```suggestion
	cleanupObjectStorageFiles := job.warehouse.GetBoolDestinationConfig(model.CleanupObjectStorageFilesSetting)
	if !cleanupObjectStorageFiles {
		return nil
	}
	destination := job.warehouse.Destination
	storageProvider := whutils.ObjectStorageType(destination.DestinationDefinition.Name, destination.Config, job.upload.UseRudderStorage)
```

Reply:
Done